crates/lsp/src/main_loop.rs

```diff
@@ -165,6 +166,9 @@ pub(crate) struct LspState {
 
     /// List of client capabilities that we care about
     pub(crate) capabilities: ResolvedClientCapabilities,
+
+    /// State used to dynamically update the log level
+    pub(crate) log_state: Option<logging::LogState>,
```

Comment:
I'm excited to move to lsp-server because I think the startup procedure there will result in us creating `GlobalState` / `LspState` _after_ we've received `InitializeParams`, meaning we will be able to get rid of `Option<>`s here and in `GlobalState`